src/components/Navigation/Pages/PageBrowse/PageBrowse.js

```diff
@@ -63,13 +64,19 @@ class PageBrowse extends SystemManagedList {
   }
 
   onBrowse = value => {
-    this.setState({ activeName: value }, () => this.updateData())
+    this.setState({ activeName: value, searchTerm: '' }, () => this.updateData())
   }
 
   onFocusChange = value => {
     this.setState({ searchFocused: value })
   }
 
+  onSearch = value => {
+    this.setState({ searchTerm: value })
+    const provider = this.state.selectedProvider.value
+    super.onSearch(provider + '/' + value)
```

Comment:
rather than calling super, can we pass in the method as a prop instead? More on this anti-pattern can be found in [this](https://github.com/facebook/react/issues/3280) discussion

Reply:
nevermind, this is merged. We'll come back to this